Frontend/ui-driver/src/Screens/RideRequestScreen/Controller.purs

```diff
@@ -0,0 +1,344 @@
+                                                                                                  Just (ScheduleBooking cardData) -> do 
+                                                                                                                                        let (BookingAPIEntity booking) = cardData.bookingDetails
+                                                                                                                                            now = EHC.getCurrentUTC ""
+                                                                                                                                            diff = runFn2 differenceBetweenTwoUTC booking.startTime now 
+                                                                                                                                        if diff < 0 then continue state else exit $ GoToRideSummary state{data{currCard = cardData.bookingDetails, fareDetails = cardData.fareDetails}}
+
+eval (OnFadeComplete _ ) state = do      
+  if not state.props.receivedResponse 
+    then do
+      continue state
+    else do
+      update state {data{
+            shimmerLoader = case state.data.shimmerLoader of
+                              AnimatedIn ->AnimatedOut
+                              AnimatingOut -> AnimatedOut
+                              a -> a
+              }
+  }
+
+eval (Notification notificationType) state = do
+    -- _ <- pure $ printLog "notificationType" notificationType
+    if (checkNotificationType notificationType ST.DRIVER_ASSIGNMENT ) then do
+      exit $ FcmNotifications notificationType state
+    else continue state
+
+eval _ state = update state
+
+
+checkNotificationType :: String -> ST.NotificationType -> Boolean
+checkNotificationType currentNotification requiredNotification = (show requiredNotification) == currentNotification
+
+myRideListTransformerProp :: Array ScheduleBooking -> Array RideCardItemState
+myRideListTransformerProp listres =
+  let resp = map ( \(ScheduleBooking ride) ->
+        let
+          (BookingAPIEntity booking)  = ride.bookingDetails
+          time = (convertUTCtoISC booking.createdAt "h:mm A")
+
+          totalAmount = Int.ceil ( booking.estimatedFare )
+
+          estimatedDuration = show $ fromMaybe 0 booking.estimatedDuration/3600
+
+          startTime = (convertUTCtoISC booking.startTime "h:mm A")
+
+          id = booking.id
+
+          date = (convertUTCtoISC booking.createdAt "DD/MM/YYYY")
+
+          distance = fromMetersToKm (fromMaybe 1 booking.estimatedDistance)
+
+          (Location fromLocation) = booking.fromLocation
+
+          srcLat = fromLocation.lat
+
+          srcLong = fromLocation.lon
+
+          (Location toLocation) =
+            fromMaybe
+              ( Location
+                  { address:
+                      LocationAddress
+                        { area: Just ""
```

Comment:
Instead of Just "" make it Nothing, we cant do case match intead of fromMaybe? createdAt can be ""